Modules/_asynciomodule.c

```diff
@@ -74,6 +129,50 @@ typedef struct {
 
     futureiterobject *fi_freelist;
     Py_ssize_t fi_freelist_len;
+
+    /* Linked-list of all tasks which are instances of asyncio.Task or subclasses
+       of it. Third party tasks implementations which don't inherit from
+       asyncio.Task are managed by the 'non_asyncio_tasks' WeakSet separately.
+       `tail` is used as sentinel to mark end of linked-list and it avoids one
+       branch in checking for empty list when adding a new task, the list is
+       initialized with `head` pointing to `tail` to mark an empty list.
+
+       Invariants:
+        * When the list is empty:
+        - asyncio_tasks.head == &asyncio_tasks.tail
+        - asyncio_tasks.head->prev == NULL
+        - asyncio_tasks.head->next == NULL
+
+        * After adding a new task 'task1':
+        - asyncio_tasks.head == task1
+        - task1->prev == &asyncio_tasks.tail
+        - task1->next == NULL
+        - asyncio_tasks.tail->next == task1
```

Comment:
tail is not a pointer

```suggestion
        - asyncio_tasks.tail.next == task1
```